layouts/partials/language-switcher.html

```diff
@@ -1,5 +1,16 @@
-<ul class="language-select">
-{{ range $.Site.Home.AllTranslations }}
-<li><a href="{{ .Permalink }}">{{ .Language.LanguageName }}</a></li>
+{{ if .IsTranslated }}
```

Comment:
There is a problem with that condition — if some post is not translated, there is no switcher at all. 
I think correct behavior should be: show switcher with links to home page with the selected language

Reply:
Yeah, great idea! Nice teamwork :)

I've implemented the fix you suggested and also refactored the code a little bit.

Let's have a look once again.